daemon/emer-daemon.c

```diff
@@ -141,7 +142,8 @@ typedef struct _EmerDaemonPrivate
 {
   guint network_send_interval;
   GQueue *upload_queue;
-  gboolean uploading;
+  /* Non-NULL iff an upload is in flight */
+  GCancellable *current_upload_cancellable;
```

Comment:
Every `GTask` in the queue has its own `GCancellable`. This field holds a new ref to the current task's cancellable:

```c
  priv->current_upload_cancellable = g_object_ref (
    g_task_get_cancellable (upload_task));
```

and, as you say, unrefs it when the task ends. But actually I think it should be cleared before returning a value to the application code. Right now the GTask returns, then the cancellable is cleared – but if the GTask's callback caused another task to be popped from the queue, the wrong cancellable would get cleared. In practice this doesn't happen – there's a connectivity check, and handle_network_monitor_can_reach is (presumably?) called back asynchronously, but I think I should make this situation impossible rather than doesn't-happen-by-accident.